test/elm/interval/test.coffee

```diff
@@ -1405,3 +1405,699 @@ describe 'Collapse', ->
+    prettyList(@msPrecPerMinute.exec(@ctx)).should.equal e
+    
+    # define MsPrecPerSecond: expand { Interval[@2018-01-01T01:00:00.000+00:00, @2018-01-01T01:00:02.000+00:00] } per second
+    e = "{ [2018-01-01T01:00:00.000+00:00, 2018-01-01T01:00:00.999+00:00], [2018-01-01T01:00:01.000+00:00, 2018-01-01T01:00:01.999+00:00] }"
+    prettyList(@msPrecPerSecond.exec(@ctx)).should.equal e
+    
+    # define MsPrecPerMillisecond: expand { Interval[@2018-01-01T01:00:00.000+00:00, @2018-01-01T01:00:00.002+00:00] } per millisecond
+    e = "{ [2018-01-01T01:00:00.000+00:00, 2018-01-01T01:00:00.000+00:00], [2018-01-01T01:00:00.001+00:00, 2018-01-01T01:00:00.001+00:00] }"
+    prettyList(@msPrecPerMillisecond.exec(@ctx)).should.equal e
+
+  it 'expands a second precision datetime', ->
+    # define SecPrecPerYear: expand { Interval[@2016-01-01T01:00:00+00:00, @2018-01-01T01:00:00+00:00] } per year
+    e = "{ [2016-01-01T00:00:00+00:00, 2016-12-31T23:59:59+00:00], [2017-01-01T00:00:00+00:00, 2017-12-31T23:59:59+00:00] }"
+    prettyList(@secPrecPerYear.exec(@ctx)).should.equal e
+
+    # define SecPrecPerMonth: expand { Interval[@2018-01-01T01:00:00+00:00, @2018-03-01T01:00:00+00:00] } per month
+    e = "{ [2018-01-01T00:00:00+00:00, 2018-01-31T23:59:59+00:00], [2018-02-01T00:00:00+00:00, 2018-02-28T23:59:59+00:00] }"
+    prettyList(@secPrecPerMonth.exec(@ctx)).should.equal e
+
+    # define SecPrecPerWeek: expand { Interval[@2018-01-01T01:00:00+00:00, @2018-01-21T01:00:00+00:00] } per week
+    e = "{ [2018-01-01T00:00:00+00:00, 2018-01-07T23:59:59+00:00], [2018-01-08T00:00:00+00:00, 2018-01-14T23:59:59+00:00] }"
+    prettyList(@secPrecPerWeek.exec(@ctx)).should.equal e
+    
+    # define SecPrecPerDay: expand { Interval[@2018-01-01T01:00:00+00:00, @2018-01-03T01:00:00+00:00] } per day
+    e = "{ [2018-01-01T00:00:00+00:00, 2018-01-01T23:59:59+00:00], [2018-01-02T00:00:00+00:00, 2018-01-02T23:59:59+00:00] }"
+    prettyList(@secPrecPerDay.exec(@ctx)).should.equal e
+    
+    # define SecPrecPerHour: expand { Interval[@2018-01-01T01:00:00+00:00, @2018-01-01T03:00:00+00:00] } per hour
+    e = "{ [2018-01-01T01:00:00+00:00, 2018-01-01T01:59:59+00:00], [2018-01-01T02:00:00+00:00, 2018-01-01T02:59:59+00:00] }"
+    prettyList(@secPrecPerHour.exec(@ctx)).should.equal e
+    
+    # define SecPrecPerMinute: expand { Interval[@2018-01-01T01:00:00+00:00, @2018-01-01T01:02:00+00:00] } per minute
+    e = "{ [2018-01-01T01:00:00+00:00, 2018-01-01T01:00:59+00:00], [2018-01-01T01:01:00+00:00, 2018-01-01T01:01:59+00:00] }"
+    prettyList(@secPrecPerMinute.exec(@ctx)).should.equal e
+    
+    # define SecPrecPerSecond: expand { Interval[@2018-01-01T01:00:00+00:00, @2018-01-01T01:00:01+00:00] } per second
+    e = "{ [2018-01-01T01:00:00+00:00, 2018-01-01T01:00:00+00:00], [2018-01-01T01:00:01+00:00, 2018-01-01T01:00:01+00:00] }"
```

Comment:
I think this is another thing we should check w/ Bryn on -- given that these intervals have imprecision.  For example, `@2018-01-01T01:00:00+00:00` represents "some moment on Jan 1, 2018 between 1:00:00.0 and 1:00:00.999" -- because the lack of milliseconds means it could be any millisecond during the specified second.  So should the result keep that same level of imprecision throughout?  Or should it be smart enough to say:
```
e = "{ [2018-01-01T01:00:00+00:00, 2018-01-01T01:00:00.999+00:00], [2018-01-01T01:00:01.0+00:00, 2018-01-01T01:00:01+00:00] }"
```

Reply:
BTW -- I think this same question would apply to all the examples where there is imprecision.  I wouldn't change anything right now, but I do think we need to reach out to Bryn for clarification.

Reply:
I believe this is satisfied by Bryn's email.